go/vt/vtctl/grpcvtctldserver/server.go

```diff
@@ -5080,16 +5083,41 @@ func (s *VtctldServer) getTopologyCell(ctx context.Context, cellPath string) (*v
 		return nil, err
 	}
 
-	if data, _, err := conn.Get(ctx, relativePath); err == nil {
-		result, err := topo.DecodeContent(relativePath, data, false)
-		if err != nil {
-			err := vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "error decoding file content for cell %s: %v", cellPath, err)
+	if version != 0 {
+		data, err := conn.GetVersion(ctx, relativePath, version)
+		if err == nil {
+			result, err := topo.DecodeContent(relativePath, data, asJSON)
```

Comment:
we can extract the repeated code that decodes the content and returns the topoCell  in the two if/else conditions into a helper.